frontend/src/metabase/dashboard/actions/sharing.ts

```diff
@@ -28,49 +28,58 @@ export const setSharing = (isSharing: boolean) => (dispatch: Dispatch) => {
 
 export const UPDATE_ENABLE_EMBEDDING =
   "metabase/dashboard/UPDATE_ENABLE_EMBEDDING";
-export const updateEnableEmbedding = createAction(
-  UPDATE_ENABLE_EMBEDDING,
-  ({ id }: DashboardIdPayload, enable_embedding: boolean) =>
-    DashboardApi.update({
-      id,
-      enable_embedding,
-    }),
-);
+export const updateEnableEmbedding = createAsyncThunk<
+  Dashboard,
+  Pick<Dashboard, "id" | "enable_embedding">
+>(UPDATE_ENABLE_EMBEDDING, async ({ id, enable_embedding }) => {
+  const response = await DashboardApi.update({
+    id,
+    enable_embedding,
+  });
+
+  return response;
+});
 
 export const UPDATE_EMBEDDING_PARAMS =
   "metabase/dashboard/UPDATE_EMBEDDING_PARAMS";
-export const updateEmbeddingParams = createAction(
-  UPDATE_EMBEDDING_PARAMS,
-  ({ id }: DashboardIdPayload, embedding_params: EmbedOptions) =>
-    DashboardApi.update({ id, embedding_params }),
-);
+export const updateEmbeddingParams = createAsyncThunk<
+  Dashboard,
+  Pick<Dashboard, "id" | "embedding_params">
+>(UPDATE_EMBEDDING_PARAMS, async ({ id, embedding_params }) => {
+  const response = await DashboardApi.update({ id, embedding_params });
+
+  return response;
+});
 
 export const CREATE_PUBLIC_LINK = "metabase/dashboard/CREATE_PUBLIC_LINK";
-export const createPublicLink = createThunkAction(
+export const createPublicLink = createAsyncThunk(
   CREATE_PUBLIC_LINK,
-  ({ id }: DashboardIdPayload) =>
-    async (dispatch: Dispatch) => {
-      const { data } = await (dispatch(
-        createDashboardPublicLink.initiate({
-          id,
-        }),
-      ) as Promise<{ data: { uuid: string }; error: unknown }>);
+  async (
+    { id }: DashboardIdPayload,
+    { dispatch, fulfillWithValue, rejectWithValue },
+  ) => {
+    const { data, error } = await dispatch(
+      createDashboardPublicLink.initiate({ id }),
+    );
+
+    if (!data || error) {
+      return rejectWithValue(error);
+    }
```

Comment:
this part looks new, are we using the error value at the moment? If not why do we add this?

Reply:
I was thinking I could include this when I was testing it but it's not adding anything right now. I'll leave it out

Reply:
ah I remember why now. TS wants me to check whether or not `data` is undefined or not, so I had an extra branch to work with in terms of a `if (!data)` branch. What do you think?

Reply:
That's better. I'd really like to only have code that we actually use. Having too much can clutter our app.